mule-apikit-module/src/main/java/org/mule/module/apikit/Router.java

```diff
@@ -97,21 +100,36 @@ public CoreEvent process(final CoreEvent event) throws MuleException {
 
   @Override
   public Publisher<CoreEvent> apply(Publisher<CoreEvent> publisher) {
-    return flatMap(publisher, event -> {
-      try {
-        Configuration config = registry.getConfiguration(getConfiguration().getName());
-        CoreEvent.Builder eventBuilder = CoreEvent.builder(event);
-        eventBuilder.addVariable(config.getOutboundHeadersMapName(), new HashMap<>());
-
-        HttpRequestAttributes attributes = ((HttpRequestAttributes) event.getMessage().getAttributes().getValue());
+    return flatMap(publisher, this::processWithExtension, this);
+  }
 
-        final CompletableFuture<Event> resultEvent = doRoute(event, config, eventBuilder, attributes);
+  private Publisher<CoreEvent> processWithExtension(CoreEvent event) {
+    try {
+      final CompletableFuture<Event> resultEvent;
+      if (configuration.isExtensionEnabled()) {
+        resultEvent =
+            configuration.getExtension()
+                .process(event, this, this.configuration.getRaml()); // "/home/jbrasca/AnypointStudio/studio-workspace/odata-extension-poc/src/main/resources/api/test.raml"
```

Comment:
I guess you accidentally added this comment, please remove it
`// "/home/jbrasca/AnypointStudio/studio-workspace/odata-extension-poc/src/main/resources/api/test.raml"`